cpp/src/io/parquet/chunk_dict.cu

```diff
@@ -78,17 +78,12 @@ struct map_insert_fn {
 struct map_find_fn {
   map_type::device_view& map;
 
-  template <typename T>
+  template <typename T, std::enable_if_t<column_device_view::has_element_accessor<T>()>* = nullptr>
```

Comment:
can we apply the same change to `map_insert_fn::operator()`?

Reply:
I found the root cause. The use of `if constexpr` with the `else` branch as `CUDF_UNREACHABLE` meant that the `auto` return type couldn't be correctly determined. I fixed this (and removed the other `return false`, which didn't have this issue because its return type is `bool` and not `auto`) in b66c4c4725fc6ac184d7c2a85170a583d0a1ded6.